### PR TITLE
Add `Table` trace.

### PR DESCRIPTION
**Pull request by baiguoname**
https://github.com/igiagkiozis/plotly/issues/148
This issue mentioned the trace.

### CODE REVIEW

**Comment by Nnamdi-sys** [comment]
Really great to see a Table Trace in the pipeline. Would also love to see a Pie Chart Trace if possible. Would love to contribute this, but don't have the requisite skill to do it yet.

**Comment by baiguoname** [comment]
> Really great to see a Table Trace in the pipeline. Would also love to see a Pie Chart Trace if possible. Would love to contribute this, but don't have the requisite skill to do it yet.

This is also first time I make a pr in plotly, after this pr is merged, I would like to make a similar Pie Chart pr.

**Comment by Nnamdi-sys** [comment]
> > Really great to see a Table Trace in the pipeline. Would also love to see a Pie Chart Trace if possible. Would love to contribute this, but don't have the requisite skill to do it yet.
> 
> This is also first time I make a pr in plotly, after this pr is merged, I would like to make a similar Pie Chart pr.

Hi @igiagkiozis, would be great to have this pull request approved in your next release. I tried it out and it works quite well. I also find it to be a very useful feature.

**Comment by kylemello** [comment]
@baiguoname I'm glad someone started work on this! 
I do think it would make more sense if `Table::new()` takes a header and a cells struct, unless I'm missing something (I am a rust noob). I think in your code currently there is no way to change the header or cells style without creating a new instance of a header or cell with the same values you used in `Table::new()` and altering the header field in the struct with `Table.header()`. It makes more sense to me to create the cell or header beforehand with the values and styles you want and pass it into the table when you're happy with it. I hope that makes sense.
I've made a fork of your repo and am happy to submit a pr to it :) Lmk what you think. Heres the link to my fork: https://github.com/kylemello/plotly

**Comment by baiguoname** [comment]
> @baiguoname I'm glad someone started work on this! I do think it would make more sense if `Table::new()` takes a header and a cells struct, unless I'm missing something (I am a rust noob). I think in your code currently there is no way to change the header or cells style without creating a new instance of a header or cell with the same values you used in `Table::new()` and altering the header field in the struct with `Table.header()`. It makes more sense to me to create the cell or header beforehand with the values and styles you want and pass it into the table when you're happy with it. I hope that makes sense. I've made a fork of your repo and am happy to submit a pr to it :) Lmk what you think. Heres the link to my fork: https://github.com/kylemello/plotly

Thanks very much, I think it's great. Excuse me, I'm still new to github, what should I do next? Should I close this commit, and you pull a new pr to this crate?

**Comment by kylemello** [comment]
> Thanks very much, I think it's great. Excuse me, I'm still new to github, what should I do next? Should I close this commit, and you pull a new pr to this crate?

Sorry for the late reply, I swear github never notifies me of anything.
No, I don't think so... I think submitting a pr onto your branch will do just fine. I think once you merge it you'll be set. I haven't done a pull request on a pull request on GitHub yet so I'm just guessing here lol. Just opened it, here's the link: https://github.com/baiguoname/plotly/pull/1

File: examples/basic_charts/src/main.rs [hunk 1]
```diff
@@ -604,6 +605,19 @@ fn basic_sankey_diagram() {
...
+fn table_chart() {
+    let trace = Table::new(
+        vec![String::from("col1"), String::from("col2")],
+        vec![
+            vec![1, 2],
+            vec![2, 3],
+        ]
+    );
+    let mut plot = Plot::new();
+    plot.add_trace(table_trace);
```
Comment (andrei-ng) [inline examples/basic_charts/src/main.rs:617]: @baiguoname  There is a typo here. This should be `trace`

**Comment by naijim** [comment]
@baiguoname hi, is this feature still able to be merged? Really looking forward to see it.

**Comment by naijim** [comment]
Is it ok for me to take over this PR and make it mergable?  @baiguoname @andrei-ng

**Comment by baiguoname** [comment]
> Is it ok for me to take over this PR and make it mergable? @baiguoname @andrei-ng

Sorry for late reply. Yes, it's ok for me.

**Comment by naijim** [comment]
> > Is it ok for me to take over this PR and make it mergable? @baiguoname @andrei-ng
> 
> Sorry for late reply. Yes, it's ok for me.

Thx. How do I take over it? I am not familiar with Github's PR. It looks like I just need to resolve merge conflict.

**Comment by andrei-ng** [comment]
> > > Is it ok for me to take over this PR and make it mergable? @baiguoname @andrei-ng
> > 
> > 
> > Sorry for late reply. Yes, it's ok for me.
> 
> Thx. How do I take over it? I am not familiar with Github's PR. It looks like I just need to resolve merge conflict.

Please do take it over :) .

My typical worflow is to add the other fork as another remote to my local clone, fetch the branch, check it out, do changes and push it back to the fork e.g.,
 ```
git remote add baiguoname git@github.com:baiguoname/plotly.git
git fetch baiguoname
git checkout -b table-pr baiguoname/master
```
... do changes
```
git push -u  table-pr HEAD:master
```

There might be other workflows outhere...

**Comment by naijim** [comment]
I am not so sure how to link the conflict-resolve PR to this one....

**Comment by naijim** [comment]
@baiguoname can you merge my branch into your PR branch?

**Comment by baiguoname** [comment]
> @baiguoname can you merge my branch into your PR branch?

Of course, but how can I do it? I didn't find any pull request or something.

**Comment by baiguoname** [comment]
> @baiguoname can you merge my branch into your PR branch?

Yes, of course. I have approved your [pr ](https://github.com/plotly/plotly.rs/pull/207#pullrequestreview-2122373174). Is there anything else I need to do?

**Comment by naijim** [comment]
@baiguoname I just update my branch to fix some CI issues. Can you approve it?

**Comment by andrei-ng** [comment]
@baiguoname and @naijim , should we close this PR in favour of https://github.com/plotly/plotly.rs/pull/207 ?

**Comment by baiguoname** [comment]
> @baiguoname and @naijim , should we close this PR in favour of #207 ?

Yes, I think so.

**Pull request closed without merging**